### PR TITLE
Authorization Provider Migration Guide

### PR DESCRIPTION
**Pull request by sshropshire**
### Summary of changes

 - Add example `AuthorizationProvider` style integration to `v4_MIGRATION_GUIDE`

 ### Checklist

 ~- [ ] Added a changelog entry~

### CODE REVIEW

File: v4_MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -99,6 +100,86 @@ In v4, we decoupled the Braintree SDK from Android to offer more integration fle
 `BraintreeFragment` has been replaced by a `Client` for each respective payment feature. 
 See the below payment method sections for examples of instantiating and using the feature clients. 
 
+## Authorization Provider
+
+When creating a `BraintreeClient`, you can provide a tokenization key, client token, or an `AuthorizationProvider`. When given an authorization provider, the SDK will fetch a client token on your behalf. This makes it possible to construct a `BraintreeClient` instance in `onCreate`.
```
Comment (sarahkoop) [inline v4_MIGRATION_GUIDE.md:105]:
```suggestion
When creating a `BraintreeClient`, you can provide a tokenization key, client token, or an `AuthorizationProvider`. When given an authorization provider, the SDK will fetch a client token on your behalf when it is needed. This makes it possible to construct a `BraintreeClient` instance in `onCreate`.
```

Comment (sarahkoop) [reply v4_MIGRATION_GUIDE.md:105]: Take it or leave it suggestion

File: v4_MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -99,6 +100,86 @@ In v4, we decoupled the Braintree SDK from Android to offer more integration fle
...
+
+**Assuming** the merchant can GET `https://www.my-api.com/client_token` and receive the following json response:
```
Comment (sarahkoop) [inline v4_MIGRATION_GUIDE.md:107]:
```suggestion
**Assuming** you have a server that supports GET `https://www.my-api.com/client_token` and receive the following json response:
```

I think we speak directly to merchants in this guide - does this make more sense?

File: v4_MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -99,6 +100,86 @@ In v4, we decoupled the Braintree SDK from Android to offer more integration fle
...
+
+  void getClientToken(@NonNull ClientTokenCallback callback) {
+    api.getClientToken(new Callback<ClientToken>() {
+      
+      @Override
+      public void success(ClientToken clientToken, Response retrofitResponse) {
+        if (retrofitResponse.getStatus() == 200) {
+          callback.onSuccess(clientToken.getValue());
+        } else {
+	  // TODO: inspect response for error reason
```
Comment (sarahkoop) [inline v4_MIGRATION_GUIDE.md:152]: Is this TODO for us or the merchant? Should this be removed?

File: v4_MIGRATION_GUIDE.md [hunk 4]
```diff
@@ -99,6 +100,86 @@ In v4, we decoupled the Braintree SDK from Android to offer more integration fle
...
+
+// ExampleAuthorizationProvider.java
+class ExampleAuthorizationProvider implements AuthorizationProvider {
+
+    private Api api = new RestAdapter.Builder()
+            .setEndpoint("https://my-api.com")
+            .build()
+            .create(Api.class);
+
+  void getClientToken(@NonNull ClientTokenCallback callback) {
```
Comment (sarahkoop) [inline v4_MIGRATION_GUIDE.md:144]: Did we change the name of this to AuthorizationCallback?

**Review by sarahkoop — APPROVED** [review]
A few nitpicks but looks good to me

**Comment by sarahkoop** [comment]
I moved this documentation into #523. I think we can close this PR - unless there's anything here I missed.

**Comment by sshropshire** [comment]
> I moved this documentation into #523. I think we can close this PR - unless there's anything here I missed.

Yeah I agree. This was preliminary. Closing now.

**Pull request closed without merging**